fiat-roles/src/main/java/com/netflix/spinnaker/fiat/providers/AccessControlledResourcePermissionSource.java

```diff
@@ -19,13 +19,16 @@
 import com.netflix.spinnaker.fiat.model.resources.Permissions;
 import com.netflix.spinnaker.fiat.model.resources.Resource;
 import java.util.Optional;
+import javax.annotation.Nonnull;
 
 public final class AccessControlledResourcePermissionSource<T extends Resource.AccessControlled>
     implements ResourcePermissionSource<T> {
 
   @Override
-  public Permissions getPermissions(T resource) {
-    return Optional.ofNullable(resource.getPermissions())
+  @Nonnull
+  public Permissions getPermissions(@Nonnull T resource) {
+    return Optional.ofNullable(resource)
```

Comment:
he marks resource `@Nonnull` then `Optional.ofNullable`'s it... trust levels showing through